tests/test_client.py

```diff
@@ -0,0 +1,129 @@
+import pytest
+
+from unittest.mock import patch
+from sdk import SDK, Endpoints, Auth0Conf
+from sdk.authentication import FakeAuth0BadAuthentication, FakeAuth0GoodAuthentication
+
+
+@patch("sdk.client.Auth0TokenProvider", FakeAuth0GoodAuthentication)
+class TestAuthSuccess:
+    group_id = "random_group_id"
+    username = "random_username"
+    password = "random_password"
+    new_core_endpoint = "random_endpoint"
+
+    @patch("sdk.client.getpass")
+    def test_module_getpass_success(self, getpass):
+        getpass.return_value = self.password
+        SDK(group_id=self.group_id, username=self.username)
+        getpass.assert_called_once()
+
+    def test_authentication_success(self):
+        SDK(group_id=self.group_id, username=self.username, password=self.password)
+
+    def test_good_token_provider(self):
+        SDK(group_id=self.group_id, token_provider=FakeAuth0GoodAuthentication)
+
+    def test_correct_endpoints(self):
+        sdk = SDK(
+            group_id=self.group_id,
+            username=self.username,
+            password=self.password,
+            endpoints=Endpoints(core=self.new_core_endpoint),
+        )
+        assert sdk._client.endpoints.core == self.new_core_endpoint
+
+    def test_correct_new_auth0(self):
+        new_auth0 = Auth0Conf(domain="new_domain")
+        SDK(
+            group_id=self.group_id,
+            username=self.username,
+            password=self.password,
+            auth0=new_auth0,
+        )
+
+
+@patch("sdk.client.Auth0TokenProvider", FakeAuth0BadAuthentication)
```

Comment:
nit: Can we rename FakeAuth0BadAuthentication to follow the same nomenclature as the test class for consistency ?

Reply:
done